### PR TITLE
Update gardenlet's base image to alpine:3.16.2

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area security
/kind enhancement

**What this PR does / why we need it**:
alpine 3.16.2 is released: [ref](https://alpinelinux.org/posts/Alpine-3.13.12-3.14.8-3.15.6-3.16.2-released.html)

**Release note**:

```other developer
gardenlet's base image is updated from `alpine:3.16.1` to `alpine:3.16.2`.
```

### CODE REVIEW

File: test/framework/resources/templates/vpntunnel-copy.yaml.tpl [hunk 1]
```diff
@@ -28,7 +28,7 @@ spec:
         volumeMounts:
         - name: source-data
           mountPath: /data
-      - image: eu.gcr.io/gardener-project/3rd/alpine:3.16.1
+      - image: eu.gcr.io/gardener-project/3rd/alpine:3.16.2
```
Comment (ialidzhikov) [inline test/framework/resources/templates/vpntunnel-copy.yaml.tpl]: `eu.gcr.io/gardener-project/3rd/alpine:3.16.2` is not multi-arch. There was a PR from @acumino  that switched all test images to be multi-arch. We don't want to regress this.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: test/framework/resources/templates/vpntunnel-copy.yaml.tpl (hunk 1 above)
Comment (oliver-goetz) [reply test/framework/resources/templates/vpntunnel-copy.yaml.tpl]: okay, I did not know that. Changed.

File: test/framework/resources/templates/simple-load-deployment.yaml.tpl [hunk 2]
```diff
@@ -14,7 +14,7 @@ spec:
         app: load
     spec:
       containers:
-      - image: eu.gcr.io/gardener-project/3rd/alpine:3.16.1
+      - image: eu.gcr.io/gardener-project/3rd/alpine:3.16.2
```
Comment (ialidzhikov) [inline test/framework/resources/templates/simple-load-deployment.yaml.tpl]: Same as https://github.com/gardener/gardener/pull/6492#discussion_r945618410.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: test/framework/resources/templates/simple-load-deployment.yaml.tpl (hunk 2 above)
Comment (oliver-goetz) [reply test/framework/resources/templates/simple-load-deployment.yaml.tpl]: 😄

**Comment by ialidzhikov** [comment]
Can I ask you to also delete `eu.gcr.io/gardener-project/3rd/alpine:3.16.2` as it is not a multi-arch image? Thanks in advance!

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/approve

**Pull request merged**